src/Source.jl

```diff
@@ -179,7 +175,7 @@ function Source(;fullpath::Union{AbstractString,IO}="",
         end
     end
     if !weakrefstrings
-        columntypes = [T <: WeakRefString ? String : T for T in columntypes]
+        columntypes = [Missings.T(T) <: WeakRefString ? substitute(T, String) : T for T in columntypes]
```

Comment:
That's where `substitute(Missing, T)` rule is required, because `Missings.T(Missing) == Union{} <: WeakRefString`.